### ISSUE TITLE
`futures` option for swapping DUSD to dTokens

### ISSUE THREAD

**Issue by thedoublejay** [DeFiCh/wallet#2394]
#### Dependencies

- [x] Required to be merged first - https://github.com/DeFiCh/jellyfish/pull/1273.
- [x] Jellyfish Future Swap Issue - https://github.com/DeFiCh/jellyfish/issues/1268#issuecomment-1085647782

Blocked by https://github.com/DeFiCh/jellyfish/issues/1355 and https://github.com/DeFiCh/ain/issues/1202

#### What would you like to be added:
Future Swap — users will be able to submit future swap orders (to swap DUSD <--> dToken at ±5% oracle price) at any time. This will be treated like auction bids, users can submit orders beforehand, but the transaction will only take place on the futures block (every mod (7*2880) === 0th block).

- [ ] Have the option to select instant swap/future swap. Instant swap is the current one that we have
![image](https://user-images.githubusercontent.com/25013382/161670802-fc98604d-ca02-4885-b1ba-6ffd6154f83b.png)

- [ ] Future Swap - Oracle Price +5%. (DUSD -> DToken) -5% (DToken -> DUSD)
![image](https://user-images.githubusercontent.com/25013382/161670893-3ad60385-6fd7-4ba1-8257-dccf97edb741.png)

- [ ] Ability to cancel pending future swaps
UI only: #2569
![image](https://user-images.githubusercontent.com/25013382/161670933-71197092-47ee-4c61-be88-4ddc58bf34f5.png)

- [ ] List all future swaps
UI only: #2570 
![image](https://user-images.githubusercontent.com/25013382/161670954-2a435d27-ee2a-47b3-ba36-81d667e31ab7.png)

- [ ] Cancel order - This will involve passcode prompt as it's a transaction.
UI only: #2571 
![image](https://user-images.githubusercontent.com/25013382/161670983-e9049081-1241-46ff-8e0c-269935704f7e.png)

**This will be the new layout, for normal swap, we will just hide the toggle

**To check, on another ticket. Do we include "future swap" amounts as locked amounts on the portfolio value?

#### Prototype
https://www.figma.com/proto/SDIpZciuxJlUf4xf8yheiY/PRDCT---Mobile-Light-Wallet?page-id=6556%3A133899&node-id=9770%3A200588&viewport=241%2C48%2C0.69&scaling=scale-down&starting-point-node-id=9770%3A200588&show-proto-sidebar=1
